[helm] Si bytes calculation always floor fix

edwintye opened the pull request:
#### What this PR does

Changes the computation of `mimir.siToBytes` to use floating point arithmetics.


<!--
Additionally, adding a multiplication factor of `0.9` to the store gateway `GOMEMLIMIT` calculation. In the issue, it was suggested the multiplication factor to be `0.8` but after some testing and looking at other projects (such as prometheus) I settled on `0.9`. Looking at the examples, it is common that the memory request and limit will be different so being a more generous number `0.9` is probably fine - some would argue a straight translation is probably fine as well which is the case in the jsonnet version. I am not too sure how exactly the jsonnet version works but from what I can tell this change does create a drift on the `GOMEMLIMIT` calculation between helm and jsonnet.
-->

The tests in ci - `compare-helm-with-jsonnet/compare-manifests` to be exact - is failing but I can't get it to work and looking at the other PRs it looks like they have been failing for a while.

#### Which issue(s) this PR fixes or relates to

Fixes #9780

#### Checklist

- [x] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

dimitarvdimitrov reviewed (changes requested):
thanks for contributing the fix! The two changes in this PR looks orthogonal to each other: fixing `siToBytes` and changing the store-gateway GOMEMLIMIT. Can you open a PR for each (perhaps repurpose this PR and open a new one)?

I'm not sure about the 0.9 multiplication of the memory request for GOMEMLIMIT. At the every least we need some comparisons running this side-by-side with the old version for example to prove the need for it.

tacole02 commented on operations/helm/charts/mimir-distributed/CHANGELOG.md:
```diff
@@ -41,6 +41,7 @@ Entries should include a reference to the Pull Request that introduced the chang
 * [BUGFIX] Fix how `fullnameOverride` is reflected in generated manifests. #9564
 * [BUGFIX] Fix `extraObjects` linting with helm lint by padding with an extra new line. #9863
 * [BUGFIX] Alertmanager: Set -server.http-idle-timeout to avoid EOF errors in ruler, also for zone aware Alertmanager #9851
+* [BUGFIX] Fix calculation of `mimir.siToBytes` and use floating point arithmetics, also multiply the memory request by 0.9 for store-gateway `GOMEMLIMIT` #10044
```
```suggestion
* [BUGFIX] Fix calculation of `mimir.siToBytes` and use floating point arithmetics. Also, multiply the memory request by 0.9 for store-gateway `GOMEMLIMIT` #10044
```

edwintye commented:
Thanks Dimitar for the suggestion. I didn't realize that the jsonnet variant didn't have a multiplication factor for `GOMEMLIMIT` and only realize while creating the PR. I have now removed that commit and updated the changelog.

chencs commented:
The CHANGELOG has just been cut to prepare for the next release. Please rebase `main` and eventually move the CHANGELOG entry added / updated in this PR to the top of the operations/helm/charts/mimir-distributed/CHANGELOG.md document. Thanks!

edwintye commented:
Thanks for the reminder. Rebased and moved.

dimitarvdimitrov reviewed (approved):
LGTM, thank you!
